git_machete/cmd.py

```diff
@@ -856,17 +856,123 @@ def __init__(self):
         pass
 
 
+def flush():
+    global branch_defs_by_sha_in_reflog, initial_log_shas_cached, remaining_log_shas_cached, remote_tracking_branches_cached, sha_by_revision_cached
```

Comment:
This should be handled in a bit different way. It looks very bash'y way.

Reply:
Yes, I'm aware of how much mess with globals I made in this script :D there's a task to tidy up all that (#27)